adapters/resetdigital/resetdigital.go

```diff
@@ -0,0 +1,307 @@
+			continue
+		}
+
+		splittedRequestData, err := processDataFromRequest(requestData, imp, bidType)
+		if err != nil {
+			errors = append(errors, err)
+			continue
+		}
+
+		requestBody, err := json.Marshal(splittedRequestData)
+
+		if err != nil {
+			errors = append(errors, err)
+			continue
+		}
+
+		requests = append(requests, &adapters.RequestData{
+			Method:  "POST",
+			Uri:     a.endpointUri,
+			Body:    requestBody,
+			Headers: getHeaders(requestData),
+			ImpIDs:  []string{imp.ID},
+		})
+	}
+
+	return requests, errors
+}
+
+func processDataFromRequest(requestData *openrtb2.BidRequest, imp openrtb2.Imp, bidType openrtb_ext.BidType) (resetDigitalRequest, error) {
+	var reqData resetDigitalRequest
+
+	if requestData.Site != nil {
+		reqData.Site.Domain = requestData.Site.Domain
+		reqData.Site.Referrer = requestData.Site.Page
+	}
+
+	rdImp := resetDigitalImp{
+		BidID: requestData.ID,
+		ImpID: imp.ID,
+	}
+
+	if bidType == openrtb_ext.BidTypeBanner && imp.Banner != nil {
+		var tempH, tempW int64
+		if imp.Banner.H != nil {
+			tempH = *imp.Banner.H
+		}
+		if imp.Banner.W != nil {
+			tempW = *imp.Banner.W
+		}
+		if tempH > 0 && tempW > 0 {
+			rdImp.MediaTypes.Banner.Sizes = append(rdImp.MediaTypes.Banner.Sizes, []int64{tempW, tempH})
+		}
+	}
+	if bidType == openrtb_ext.BidTypeVideo && imp.Video != nil {
+		var tempH, tempW int64
+		if imp.Video.H != nil {
+			tempH = *imp.Video.H
+		}
+		if imp.Video.W != nil {
+			tempW = *imp.Video.W
+		}
+		if tempH > 0 && tempW > 0 {
+			rdImp.MediaTypes.Video.Sizes = append(rdImp.MediaTypes.Video.Sizes, []int64{tempW, tempH})
+		}
+		if imp.Video.MIMEs != nil {
+			rdImp.MediaTypes.Video.Mimes = append(rdImp.MediaTypes.Video.Mimes, imp.Video.MIMEs...)
+		}
+	}
+	if bidType == openrtb_ext.BidTypeAudio && imp.Audio != nil && imp.Audio.MIMEs != nil {
+		rdImp.MediaTypes.Audio.Mimes = append(rdImp.MediaTypes.Audio.Mimes, imp.Audio.MIMEs...)
+	}
+
+	var bidderExt adapters.ExtImpBidder
+	var resetDigitalExt openrtb_ext.ImpExtResetDigital
+
+	if err := json.Unmarshal(imp.Ext, &bidderExt); err != nil {
+		return resetDigitalRequest{}, err
+	}
+	if err := json.Unmarshal(bidderExt.Bidder, &resetDigitalExt); err != nil {
+		return resetDigitalRequest{}, err
+	}
+	rdImp.ZoneID.PlacementID = resetDigitalExt.PlacementID
+
+	reqData.Imps = append(reqData.Imps, rdImp)
+
+	return reqData, nil
+}
+
+func (a *adapter) MakeBids(request *openrtb2.BidRequest, requestData *adapters.RequestData, responseData *adapters.ResponseData) (*adapters.BidderResponse, []error) {
+	if adapters.IsResponseStatusCodeNoContent(responseData) {
+		return nil, nil
+	}
+
+	if err := adapters.CheckResponseStatusCodeForErrors(responseData); err != nil {
+		return nil, []error{err}
+	}
+
+	var response resetDigitalBidResponse
+	if err := json.Unmarshal(responseData.Body, &response); err != nil {
+		return nil, []error{err}
+	}
+
+	bidResponse := adapters.NewBidderResponseWithBidsCapacity(len(request.Imp))
```

Comment:
It looks like above you are only sending one imp per request ... so this seems incorrect. This function is being passed the response to one of your above generated requests, so will likely have just one bid. More if you often send more than one bid per imp of course. PBS core then merges all the bids gotten by calling this function for each request/response that was generated for the incoming request.

Reply:
You are right, but at the same time, doesn't the code still work fine?